src/ILogger/ApplicationInsightsLoggerProvider.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ApplicationInsightsLoggerProvider.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2013
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.Extensions.Logging.ApplicationInsights
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.ApplicationInsights;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.Extensions.Options;
+
+    /// <summary>
+    /// Application insights logger provider.
+    /// </summary>
+    /// <seealso cref="Microsoft.Extensions.Logging.ILoggerProvider" />
+    /// <seealso cref="Microsoft.Extensions.Logging.ISupportExternalScope" />
+    [ProviderAlias("ApplicationInsights")]
+    public class ApplicationInsightsLoggerProvider : ILoggerProvider, ISupportExternalScope
+    {
+        /// <summary>
+        /// The application insights logger options.
+        /// </summary>
+        private readonly ApplicationInsightsLoggerOptions applicationInsightsLoggerOptions;
+
+        /// <summary>
+        /// The telemetry client to be used to log messages to Application Insights.
+        /// </summary>
+        private readonly TelemetryClient telemetryClient;
+
+        /// <summary>
+        /// The collection of application insights loggers stored against categoryName.
+        /// categoryName -> <see cref="ApplicationInsightsLogger"/> map.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, ApplicationInsightsLogger> applicationInsightsLoggers
+            = new ConcurrentDictionary<string, ApplicationInsightsLogger>();
```

Comment:
Why do you need this at all?

Reply:
One option is to let Provider create a new logger every time and use that (Transient) pattern. Or hold on to instances in general and keep reusing them. I looked at [ConsoleLogger](https://github.com/aspnet/Logging/blob/master/src/Microsoft.Extensions.Logging.Console/ConsoleLoggerProvider.cs) and saw the pattern of holding on to instances and followed the same pattern here.

There is an additional assignement of SDK version is Logger as well, however that can be pulled in Provider.

Reply:
If it wasn't for option reload we wouldn't hold onto loggers in console logger either. `LoggerFactory` would cache `ILogger` instances for you

Reply:
And `SetScopeProvider` is called before any calls to `CreateLogger` so you are safe there too.

Reply:
Hmm, alright in that case I will remove caching logic inside the SDK.